build-test.sh

```diff
@@ -3,83 +3,171 @@
+  # Set variables
+  local url=http://localhost:$port/consensus
+  local cmd="until wget --user-agent='Sia-Agent' -q -O - $url
     do
       sleep 1
     done"
+  
+  # execute
+  doing "Waiting for consensus at $url"
+  timeout $timeout sh -c "$cmd"
+  finished
 }
 
+###############################################################################
+# wait_for_renter_upload_ready waits until the renter ant is upload ready.
+# Globals: none
+# Parameters:
+# - $1: Timeout to wait for the renter to become upload ready
+# - $2: API port of the renter ant to be checked
+###############################################################################
+wait_for_renter_upload_ready() {
+  # Parameters
+  local timeout=$1
+  local port=$2
+  
+  # Set variables
+  local url=http://localhost:$port/renter/uploadready
+  local cmd="until wget --user-agent='Sia-Agent' -q -O - $url | grep '\"ready\":true' 
+    do
+      sleep 1
+    done"
+  
+  # execute
+  doing "Waiting for renter to become upload ready at $url"
+  timeout $timeout sh -c "$cmd"
+  finished
+}
+
+# Remove test containers before we start
 cleanup_test_containers
 
+# Iterate over all Dockerfiles
 for DIR in ./
 do
+  # Build the image
   docker build \
     --no-cache \
     --tag sia-ant-farm-image-test \
     -f $DIR/Dockerfile \
     .
 
-  export DUMMY_DATA_DIR=$(mktemp -d)
-
   # Test with a single published port
   # Run container in detached state
+  DUMMY_DATA_DIR=$(mktemp -d)
+  doing "Starting test container"
   docker run \
     --detach \
     --publish 127.0.0.1:9988:9980 \
     --volume "${DUMMY_DATA_DIR}:/sia-antfarm/data" \
     --name sia-ant-farm-test-container \
     sia-ant-farm-image-test
+  finished
 
   # Wait till API (/consensus) is accessible
-  echo "Get consensus..."
-  curl_loop=$(curl_loop "http://localhost:9988/consensus")
-  timeout 120 sh -c "$curl_loop"
-  echo "Got consensus successfully"
+  wait_for_consensus 120 9988
 
-  docker rm -f sia-ant-farm-test-container
+  # Wait for renter to become upload ready
+  wait_for_renter_upload_ready 300 9988
+
+  # Remove test container
+  cleanup_test_containers
 
   # Test with 2 published ports
+  # Run container in detached state
+  DUMMY_DATA_DIR=$(mktemp -d)
+  doing "Starting test container 2"
   docker run \
     --detach \
     --publish 127.0.0.1:9988:9980 \
     --publish 127.0.0.1:10988:10980 \
     --volume "${DUMMY_DATA_DIR}:/sia-antfarm/data" \
     --volume "$(pwd)/config:/sia-antfarm/config" \
     --env CONFIG=config/basic-renter-5-hosts-2-api-ports-docker.json \
-    --name sia-ant-farm-test-2-apis-container \
+    --name sia-ant-farm-test-container \
     sia-ant-farm-image-test
+  finished
   
   # Wait till both APIs (/consensus) are accessible
-  echo "Get consensus..."
-  curl_loop2=$(curl_loop "http://localhost:10988/consensus")
-  timeout 120 sh -c "$curl_loop" && timeout 10 sh -c "$curl_loop2"
-  echo "Got consensus successfully"
+  wait_for_consensus 120 9988
+  wait_for_consensus 10 10988
```

Comment:
(nit) 9988 and 10988 are unusual ports, specifically mapped here. It is usually helpful to assign those to descriptively named constants. OTOH, you are consistent with your approach (9980 -> 9988, 10980 -> 10988), so it might be fine to not do that.

Reply:
Updated the test script in the next PR. The first test uses standard renter port, the second test check forwarding to a non-standard renter port.